Autopilot/PathManager/Src/takeoffMode/takeoffModeStages.cpp

```diff
@@ -0,0 +1,181 @@
+#include "takeoffModeStages.hpp"
+#include "cruisingMode.hpp"
+#include "SensorFusion.hpp"
+
+/***********************************************************************************************************************
+ * Static Member Variable Declarations
+ **********************************************************************************************************************/
+
+_LandingTakeoffInput TakeoffRollStage::_input;
+_LandingTakeoffOutput TakeoffRollStage::_output;
+_WaypointManager_Data_Out TakeoffRollStage::_waypoint_output;
+_WaypointStatus TakeoffRollStage::_waypoint_status;
+_PathData* TakeoffRollStage::_current_location;
+_PathData* TakeoffRollStage::_path_array[1];
+WaypointManager TakeoffRollStage::_takeoff_path;
+_PathData TakeoffRollStage::_takeoff_point;
+
+_LandingTakeoffInput TakeoffClimbStage::_input;
+_LandingTakeoffOutput TakeoffClimbStage::_output;
+_WaypointManager_Data_In TakeoffClimbStage::_waypoint_input;
+_WaypointManager_Data_Out TakeoffClimbStage::_waypoint_output;
+
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+void TakeoffRollStage::execute(TakeoffMode* takeoff_mode) {
+    // Reset output data
+    takeoff_mode->resetPassby(&_output.controlDetails);
+
+    //load in sensor fusion data and telemtry data into input structure
+    Telemetry_PIGO_t telem_data = takeoff_mode->getTelemetryData();
+    _input.telemetryData = &telem_data;
+
+    SFOutput_t sf_data;
+    #ifdef UNIT_TESTING
+        sf_data.airspeed = 15; // So we transition to climb stage
+    #else
+        sf_data = takeoff_mode->getSensorFusionData();
+    #endif
+    _input.sensorOutput = &sf_data;
+
+    //max throttle
+    _output.controlDetails.throttlePassby = true;
+    _output.controlDetails.throttlePercent = 100;
+
+    //making sure landing points are only made once
+    if (!takeoff_mode->getMadeTakeoffPoints()) {
+        //creating waypoints
+        _takeoff_point = LandingTakeoffManager::createTakeoffWaypoint(_input.sensorOutput->latitude, _input.sensorOutput->longitude, _input.sensorOutput->altitude, _input.telemetryData->takeoffDirectionHeading);
+        _path_array[0] = _takeoff_path.initialize_waypoint(_takeoff_point.longitude, _takeoff_point.latitude, _takeoff_point.altitude, PATH_FOLLOW);
+        
+        //10 meters is added to the altitude of the currentLocation waypoint so that is is not ground level
+        _current_location = _takeoff_path.initialize_waypoint(_input.sensorOutput->longitude, _input.sensorOutput->latitude, (_input.sensorOutput->altitude + 10), HOLD_WAYPOINT);
```

Comment:
why specifically 10 meters? Can this be made adjustable

Reply:
@NixonCTChan

Reply:
yeah we should, @Pop0097 could you make a const and set it to 10 meters? Could call it kTakeoffAltitudeOffset.